tests/unit/admin/test-class-wc-rest-payments-settings-controller.php

```diff
@@ -255,6 +266,30 @@ public function test_get_settings_fails_if_user_cannot_manage_woocommerce() {
 		remove_filter( 'user_has_cap', $cb );
 	}
 
+	public function test_get_settings_without_error_when_faulty_enabled_payment_methods() {
+		$this->upe_gateway->update_option(
+			'upe_available_payment_method_ids',
+			[
+				'card',
+				'au_becs_debit',
+				'bancontact',
+				'eps',
+				'giropay',
+				'ideal',
+				'sofort',
+				'sepa_debit',
+				'p24',
+			]
+		);
+
+		$request = new WP_REST_Request();
+		$request->set_param( 'enabled_payment_method_ids', [ 'card', 'link' ] );
+
+		$response = $this->upe_controller->get_settings( $request );
+
+		$this->assertEquals( [ 'card' ], $response->get_data()['enabled_payment_method_ids'] );
```

Comment:
Also constant + let's use `assertSame` instead of `assertEquals`. It's more strict towards types.

Reply:
I have changed this 👍 . Thanks for sharing!

Reply:
Marking as resolved.